sqlglot/lineage.py

```diff
@@ -276,8 +276,81 @@ def to_node(
                 reference_node_name=reference_node_name,
                 trim_selects=trim_selects,
             )
+
+        elif scope.pivots:
+            pivot = next(
+                (p for p in scope.pivots if p.alias_or_name == c.table and not p.unpivot), None
+            )
+            if pivot and isinstance(pivot.args.get("columns"), list):
+                # The source is a pivot operation, so we need to trace back to the aggregated column
+                pivot_column_mapping = {}
+                pivot_aggs_count = len(pivot.expressions)
+                columns = pivot.args["columns"]
+                columns_count = len(columns)
+
+                if c.alias_or_name not in [i.this for i in columns]:
+                    # The column is not in the pivot, so it must be an implicit column of the pivoted source.
+                    table = pivot.parent.name
+                    source = scope.sources.get(table)
+                    if isinstance(source, Scope):
+                        to_node(
+                            c.name,
+                            scope=source,
+                            scope_name=table,
+                            dialect=dialect,
+                            upstream=node,
+                            source_name=source_names.get(table) or source_name,
+                            reference_node_name=reference_node_name,
+                            trim_selects=trim_selects,
+                        )
```

Comment:
It looks like there's some repetition in these `to_node` / `Node(...)` (below) calls. Is it possible to refactor this so we don't have to copy-paste the same kwargs every time?

Reply:
I refactored it with the list of `downstream_columns` which is filled either by the aggregations or the implicit source column. After that the `to_node`/`Node()` part is called once